pde/org.eclipse.jdt.ls.importer.pde/src/org/eclipse/jdt/ls/importer/pde/internal/JunitLaunchConfiguration.java

```diff
@@ -81,6 +82,10 @@ public Map<String, String> toValueMap() {
 		valueMap.put("jreContainer", jreContainer);
 		valueMap.put("testProject", testProject);
 		valueMap.put("testBundle", testBundle);
+		valueMap.put("useUIThread", String.valueOf(useUIThread));
+		if (useUIThread) {
+			valueMap.put("vmArgs", "-XstartOnFirstThread");
```

Comment:
When I enabled `useUIThread` setting, the PDE JUnit launcher would fail due to "Unrecognized option: -XstartOnFirstThread".

Reply:
Was it a UI test in module like jdt.ui ?

Reply:
I tested the PR using JUnit test in eclipse.jdt.ls project, it just cannot recognize this vmArgs.

What's the purpose of this argument?

Reply:
These arguments only work with UI unit tests, the UI unit tests are marked in eclipse PluginJunit launcher to use UI Thread. When that is ticked these parameters are added. Without them the UI thread doesn’t work properly.

Reply:
But exactly what they do is something I have lot investigated. May be checking in platform discussions might give some explanation

Reply:
> These arguments only work with UI unit tests, the UI unit tests are marked in eclipse PluginJunit launcher to use UI Thread. When that is ticked these parameters are added. Without them the UI thread doesn’t work properly.

I tried to run one plugin test from `org.eclipse.jdt.ui.tests` module using Eclipse 20221201-1913, here is the generated launch template by Eclipse. I see it added vmArgs with `-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true`, no `-XstartOnFirstThread` there. 

What project and what Eclipse are you using?

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<launchConfiguration type="org.eclipse.pde.ui.JunitLaunchConfig">
    <booleanAttribute key="append.args" value="true"/>
    <booleanAttribute key="askclear" value="false"/>
    <booleanAttribute key="automaticAdd" value="true"/>
    <booleanAttribute key="automaticValidate" value="false"/>
    <stringAttribute key="bootstrap" value=""/>
    <stringAttribute key="checked" value="[NONE]"/>
    <booleanAttribute key="clearConfig" value="true"/>
    <booleanAttribute key="clearws" value="true"/>
    <booleanAttribute key="clearwslog" value="false"/>
    <stringAttribute key="configLocation" value="${workspace_loc}/.metadata/.plugins/org.eclipse.pde.core/pde-junit"/>
    <booleanAttribute key="default" value="true"/>
    <booleanAttribute key="includeOptional" value="true"/>
    <stringAttribute key="location" value="${workspace_loc}/../junit-workspace"/>
    <booleanAttribute key="org.eclipse.debug.core.ATTR_FORCE_SYSTEM_CONSOLE_ENCODING" value="false"/>
    <listAttribute key="org.eclipse.debug.core.MAPPED_RESOURCE_PATHS">
        <listEntry value="/org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/dialogs/DialogsTest.java"/>
    </listAttribute>
    <listAttribute key="org.eclipse.debug.core.MAPPED_RESOURCE_TYPES">
        <listEntry value="1"/>
    </listAttribute>
    <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
    <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
    <stringAttribute key="org.eclipse.jdt.junit.TESTNAME" value=""/>
    <stringAttribute key="org.eclipse.jdt.junit.TEST_KIND" value="org.eclipse.jdt.junit.loader.junit5"/>
    <booleanAttribute key="org.eclipse.jdt.launching.ATTR_ATTR_USE_ARGFILE" value="false"/>
    <stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
    <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.eclipse.jdt.ui.tests.dialogs.DialogsTest"/>
    <stringAttribute key="org.eclipse.jdt.launching.PROGRAM_ARGUMENTS" value="-os ${target.os} -ws ${target.ws} -arch ${target.arch} -nl ${target.nl} -consoleLog"/>
    <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="org.eclipse.jdt.ui.tests"/>
    <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.pde.ui.workbenchClasspathProvider"/>
    <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true"/>
    <stringAttribute key="pde.version" value="3.3"/>
    <stringAttribute key="product" value="org.eclipse.platform.ide"/>
    <booleanAttribute key="run_in_ui_thread" value="true"/>
    <booleanAttribute key="show_selected_only" value="false"/>
    <booleanAttribute key="tracing" value="false"/>
    <booleanAttribute key="useCustomFeatures" value="false"/>
    <booleanAttribute key="useDefaultConfig" value="true"/>
    <booleanAttribute key="useDefaultConfigArea" value="false"/>
    <booleanAttribute key="useProduct" value="true"/>
</launchConfiguration>

```

Reply:
Did you mark it as use UI Thread ? i can try removing it again and see.

Reply:
Yes, you can see the attribute "run_in_ui_thread" is true in the launch configuration.

Reply:
The paramater is needed for MacOS it seems, when it is not present this warning is shown

`***WARNING: Display must be created on main thread due to Cocoa restrictions. Use vmarg -XstartOnFirstThread`

Reply:
@testforstephen can we merge this at this state ?

Reply:
@testforstephen  Do you think we can merge this as it is, or should I change the code to add `-XstartOnFirstThread` by checking the platform ?

Reply:
It reports errors when enabling uiThread test in Windows.

![image](https://github.com/testforstephen/vscode-pde/assets/14052197/b13ac6a1-f54e-4033-ba66-82791f15deea)

Since the template for uiThread contains a `vmArgs` variable, you should at least assign it an empty value (e.g. `valueMap.put("vmArgs", "");`) for other platforms.

Also, I tested your commit by using eclipse.jdt.ls plug-in test in windows, I never make it work with uiThread enabled. It reports lots of exceptions in logs.

If you verify it works for macOS, I can merge it for you once you fixed the vmArgs issue.

Reply:
Please check now